fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPageState.kt

```diff
@@ -5,27 +5,29 @@
 package org.mozilla.fenix.onboarding.view
 
 import androidx.annotation.DrawableRes
-import org.mozilla.fenix.compose.LinkTextState
+import org.mozilla.fenix.settings.SupportUtils
 
 /**
  * Model containing data for [OnboardingPage].
  *
  * @property imageRes [DrawableRes] displayed on the page.
  * @property title [String] title of the page.
  * @property description [String] description of the page.
- * @property linkTextState [LinkTextState] part of description text with a link.
  * @property primaryButton [Action] action for the primary button.
  * @property secondaryButton [Action] action for the secondary button.
  * @property onRecordImpressionEvent Callback for recording impression event.
+ * @property onPrivacyPolicyClick action for privacy policy.
+ * @property privacyPolicyUrl url to open when user taps on privacy policy.
  */
 data class OnboardingPageState(
     @DrawableRes val imageRes: Int,
     val title: String,
     val description: String,
-    val linkTextState: LinkTextState? = null,
     val primaryButton: Action,
     val secondaryButton: Action? = null,
     val onRecordImpressionEvent: () -> Unit = {},
+    val onPrivacyPolicyClick: (String) -> Unit = {},
+    val privacyPolicyUrl: String = SupportUtils.getMozillaPageUrl(SupportUtils.MozillaPage.PRIVATE_NOTICE),
```

Comment:
I think we should revisit this approach. These shouldn't be available on each page. If we are separating description and "caption", we can add a nullable caption model which contains the String and `LinkTextState`. We should only update the mapper logic where the first page has the `caption`. Also, the UI doesn't need to check the index to display the link, just check if the caption with link is available and display it.

Reply:
(Thanks for the offline explanation) Agreed, it will make things cleaner.  I'll update the implementation.